liblocation/src/main/java/com/mapbox/android/core/location/MapboxFusedLocationEngineImpl.java

```diff
@@ -0,0 +1,160 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import static com.mapbox.android.core.location.Utils.isBetterLocation;
+
+/**
```

Comment:
I believe [LocationEngine](https://github.com/mapbox/mapbox-events-android/blob/de572d7f784141a5d57b65851e6fdce1ab3c9d32/liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineImpl.java) method `getLocationListener` is a typo and its actually `set...`.

Reply:
Also, should [AbstractLocationEngineImpl](https://github.com/mapbox/mapbox-events-android/blob/de572d7f784141a5d57b65851e6fdce1ab3c9d32/liblocation/src/main/java/com/mapbox/android/core/location/AbstractLocationEngineImpl.java) call `destroyListener` from `unmapLocationListener` when removing it from the list?

Reply:
@yunikkk you actually got me thinking there - thanks for that 👍 I've found a way to get rig of destroy completely by moving removal logic upwards in the stack. hope it looks a bit cleaner.

Reply:
Hmm and where I can find `remove` logic now? Dont see it in this PR...

Reply:
Ahh, it's `removeLocationUpdates`, found it

Reply:
So here I was just surprised that it was destroyed from `map` be not from `unmap`, now it's clear

Reply:
The follow up question is - how are `listeners` used here? First I thought that all the location updates will be sent to listeners but that seems not the case though...

Reply:
i'm keeping ref to `callback` to report back `fused` location. not all updates will be sent to listeners potentially but only the ones that are qualified as `isBetterLocation` compared to last known location.

Reply:
Yeah, `callback` is clear, I tried to understand what is the point to keep list of all callbacks in `listeners` field in [AbstractLocationEngine](https://github.com/mapbox/mapbox-events-android/blob/de572d7f784141a5d57b65851e6fdce1ab3c9d32/liblocation/src/main/java/com/mapbox/android/core/location/AbstractLocationEngineImpl.java)

Reply:
@yunikkk initially listeners field was created to keep track of unique `LocationEngineListener` callbacks, since technically user can create multiple requests and choose to handle updates from them with different callbacks, which could be mostly the case for Android location engine (not so much for Google Play version of fused provider), but since now we have our fused version of engine, this use case become less valuable. So you brought up a really good point, perhaps it'd make sense to simplify listener logic in near future.